src/checkers/inference/DefaultSlotManager.java

```diff
@@ -424,13 +403,43 @@ public SourceVariableSlot createSourceVariableSlot(AnnotationLocation location,
             throw new BugInCF("Unable to find default annotation for location " + location);
         }
 
-        AnnotationMirror realAnnotation = null;
-        if (tree != null) {
-            realAnnotation = this.defaultAnnotationsCache.get(tree);
-            if (realAnnotation == null) {
-                // If its default type can't be found in the cache, we can
-                // fallback to the simplest method.
-                realAnnotation = realTypeFactory.getAnnotatedType(tree).getAnnotationInHierarchy(this.realTop);
+        if (!defaultAnnotationsCache.containsKey(tree)) {
+            // If cache misses, we check if the top level class has changed.
+            ClassTree topLevelClass = InferenceMain.getInstance().getVisitor().getCurrentTopLevelClass();
```

Comment:
Instead of `getVisitor()` can you use `getRealChecker()` and avoid having to add `getCurrentTopLevelClass()` to both the checker and visitor?

Reply:
Ah, we would need access to the `inferenceChecker`, not the `realChecker` field... so can you think whether the changes should go into `InferrableChecker`/`BaseInferrableChecker` instead of into `InferenceChecker`?